### ISSUE TITLE
Question: What is the correct way for setting values in an alt store?

### ISSUE THREAD

**Issue by weblogixx** [goatslacker/alt#595]
Hi @goatslacker,

first of all, let me thank you for this software. Alt really rocks and makes it much easier to create flux based applications.

I am a little bit confused about the different ways of setting values for stores. On your [example pages](http://alt.js.org/guide/store/), the data is set in the following way (which does work quite well):

```javascript
class ExampleStore {
  constructor() {
    this.something = 'x';
  }

  handleSetX(data) {
    this.something = data;
  }
}
```

This is the way I am currently using alt. Sometimes I also do multiple changes like

```javascript
class ExampleStore {
  handleMultiple(data) {
    this.x = data.x;
    this.y = data.y;
  }
}
```

I have also read in some comments that the whole state should reside in single keys on the store class, but in a defined state property named ```state```:

```javascript
class ExampleStore {
  constructor() {
    this.state = {
      a: 'b'
    };
  }
}
```

In these cases, it is also said that the method ```setState()``` should be used to update the state changes. 

So my question is: What is the fundamental difference between those two assigns? Both seem to work without problems and are easily testable. However, in the getting started section the first variant is used, not hinting on the setState one.

**Comment by jdlehman** [comment]
Hey @weblogixx, both ways are completely valid and I would say that you can use what feels right based on your personal preference.

That said, `setState` was introduced to stores for a few reasons.
- to make the API more React-like as most people use alt with React
- putting the store state on a key we know about, `state`, we can create [utilities](https://github.com/altjs/utils) that provide more functionality without having to provide configuration options for the shape of the store data

So tldr; use whichever feels right to you, they both will get the job done, but storing store data on `this.state` is probably best if you plan on using utilities like the [`immutable util`](https://github.com/altjs/utils/blob/master/src/ImmutableUtil.js)
